### PR TITLE
Upgrade 'azure-mgmt-compute' to '28.0.0'

### PR DESCRIPTION
**Pull request by SRIKKANTH**
'azure-mgmt-compute == 28.0.0' is the lowest version where we can read disk_controller_type of VMs.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -40,7 +40,7 @@ aws = [
 
 azure = [
     "azure-identity ~= 1.9.0",
-    "azure-mgmt-compute ~= 26.1.0",
+    "azure-mgmt-compute ~= 28.0.0",
```
Comment (squirrelsc) [inline pyproject.toml:43]: make this PR minimal to upgrading and fixing related places. If you want to include other small changes in the same PR, put them into separated commits.

Comment (SRIKKANTH) [reply pyproject.toml:43]: All the changes as part of this PR are flake8 error fixes. There are no extra changes included in this.

File: lisa/sut_orchestrator/azure/platform_.py [hunk 2]
```diff
@@ -2259,32 +2278,50 @@ def _get_vhd_os_disk_size(self, blob_url: str) -> int:
         assert properties.size, f"fail to get blob size of {blob_url}"
         # Azure requires only megabyte alignment of vhds, round size up
         # for cases where the size is megabyte aligned
-        return math.ceil(properties.size / 1024 / 1024 / 1024)
+        vhd_os_disk_size = math.ceil(properties.size / 1024 / 1024 / 1024)
+        assert isinstance(
+            vhd_os_disk_size, int
+        ), "'vhd_os_disk_size' is not of type 'int'"
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: Refer to other `isinstance`, it should output the actual type.

File: lisa/sut_orchestrator/azure/platform_.py [hunk 3]
```diff
@@ -2117,12 +2137,11 @@ def _generate_data_disks(
...
             # we have to inject below part when dataDisks section added in
-            #  arm template,
-            # otherwise will see below exception:
-            # deployment failed: InvalidParameter: StorageProfile.dataDisks.lun
-            #  does not have required value(s) for image specified in
-            #  storage profile.
-            if marketplace:
+            # arm template, otherwise will see below exception:
+            #   deployment failed: InvalidParameter: StorageProfile.dataDisks.lun
+            #     does not have required value(s) for image specified in
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py:2142]: Fixes indentations, when the comments are reformatted.

File: lisa/sut_orchestrator/azure/features.py [hunk 4]
```diff
@@ -2204,11 +2207,13 @@ def create_or_update(
 
         if protected_settings:
             add_secret(
-                str(extension_parameters.as_dict()["protected_settings"]),
+                str(extension_parameters.protected_settings),
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py:2210]: Like this change is not related to azure upgrading, and potentially regression. Make sure you have exact tool versions.

**Review by squirrelsc — APPROVED** [review]

**Comment by squirrelsc** [comment]
@LiliDeng LGTM
